src/pool-bin/libraries/math/PackedUint128Math.sol

```diff
@@ -209,17 +212,21 @@ library PackedUint128Math {
 
     /// @dev given amount and protocolFee, calculate and return external protocol fee amt
     /// @param amount encoded bytes with (x, y)
-    /// @param fee protocolFee
-    function getExternalFeeAmt(bytes32 amount, uint16 fee) internal pure returns (bytes32 z) {
-        if (fee == 0) return 0;
+    /// @param protocolFee protocolFee
+    function getExternalFeeAmt(bytes32 amount, uint24 protocolFee) internal pure returns (bytes32 z) {
+        if (protocolFee == 0) return 0;
 
         (uint128 amountX, uint128 amountY) = decode(amount);
 
-        uint16 fee0 = fee % 256;
-        uint16 fee1 = fee >> 8;
+        uint16 fee0 = protocolFee.getZeroForOneFee();
+        uint16 fee1 = protocolFee.getOneForZeroFee();
 
-        uint128 feeForX = fee0 == 0 ? 0 : amountX / fee0;
-        uint128 feeForY = fee1 == 0 ? 0 : amountY / fee1;
+        uint128 feeForX;
+        uint128 feeForY;
+        unchecked {
```

Comment:
double check:

is it safe to use `unchecked` here ? I didn't see any overflow / underflow  case

Reply:
looks good to me too. 

maybe can add a comment here for us to double check later since we added this unchecked optimization here eg.
```
// todo: double check on this unchecked condition
```

Reply:
LGTM as well